titus-server-master/src/main/java/com/netflix/titus/master/kubernetes/pod/resourcepool/MixedSchedulerResourcePoolResolver.java

```diff
@@ -0,0 +1,70 @@
+package com.netflix.titus.master.kubernetes.pod.resourcepool;
+
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+
+import com.netflix.titus.api.jobmanager.model.job.ContainerResources;
+import com.netflix.titus.api.jobmanager.model.job.Job;
+import com.netflix.titus.api.jobmanager.model.job.Task;
+import com.netflix.titus.master.kubernetes.pod.KubePodConfiguration;
+import com.netflix.titus.master.service.management.ApplicationSlaManagementService;
+
+public class MixedSchedulerResourcePoolResolver implements PodResourcePoolResolver {
+
+    private final KubePodConfiguration configuration;
+
+    public MixedSchedulerResourcePoolResolver(KubePodConfiguration configuration) {
+        this.configuration = configuration;
+    }
+
+
+    @Override
+    public List<ResourcePoolAssignment> resolve(Job<?> job, Task task) {
+        if (!configuration.isMixedSchedulingEnabled()) {
+            return Collections.emptyList();
+        }
+        ResourcePoolAssignment elasticAssignment = ResourcePoolAssignment.newBuilder()
+                .withResourcePoolName(PodResourcePoolResolvers.RESOURCE_POOL_ELASTIC)
+                .withRule("Mixed scheduling pool assignment " + PodResourcePoolResolvers.RESOURCE_POOL_ELASTIC)
+                .build();
+        ResourcePoolAssignment reservedAssignment = ResourcePoolAssignment.newBuilder()
+                .withResourcePoolName(PodResourcePoolResolvers.RESOURCE_POOL_RESERVED)
+                .withRule("Mixed scheduling pool assignment " + PodResourcePoolResolvers.RESOURCE_POOL_RESERVED)
+                .build();
+
+        String preferredPool = calculatePreferredPool(job);
+        if (preferredPool == PodResourcePoolResolvers.RESOURCE_POOL_ELASTIC) {
+            elasticAssignment.setPreferredResourcePoolName(PodResourcePoolResolvers.RESOURCE_POOL_ELASTIC);
+            reservedAssignment.setPreferredResourcePoolName(PodResourcePoolResolvers.RESOURCE_POOL_ELASTIC);
+        } else if (preferredPool == PodResourcePoolResolvers.RESOURCE_POOL_RESERVED) {
+            reservedAssignment.setPreferredResourcePoolName(PodResourcePoolResolvers.RESOURCE_POOL_RESERVED);
+            elasticAssignment.setPreferredResourcePoolName(PodResourcePoolResolvers.RESOURCE_POOL_RESERVED);
+        }
+        return Arrays.asList(elasticAssignment, reservedAssignment);
+    }
+
+    /**
+     * calculatePreferredPool looks at a job and picks the "best" pool for the job,
+     * which is a function of its resources, *not* its capacity group!
+     * In a mixed scheduling world, we prefer to put jobs based on where they might fit best
+     * based on their ratio, and not due to a policy.
+     */
+    private String calculatePreferredPool(Job<?> job) {
```

Comment:
Future-proofing idea: return a sorted list of the preferred resource pools.

Reply:
Ive updated this to be a bool instead